### DISCUSSION TITLE
When NPT ensemble press is set to 1 in model_devi_jobs, the volume continues to increase until an error occurs

### DISCUSSION THREAD

**Discussion by Coldwater-fish** [deepmodeling/dpgen#1253] [Q&A]
I have tried CH4 and Si system. When the NPT ensemble press is set small (eg. less than 10) in model_devi_jobs, the volume continues to increase in md process until an lammps error occurs.
The lammps error is as follows:
> /home/web_server/deepmd-kit/bin/lmp: line 11: 1373361 Floating point exception(core dumped) /home/web_server/deepmd-kit/bin/_lmp "$@"

My question is whether the pressure can be calculated accurately in model_devi part? If not, does it make sense to use NPT ensemble. Actually, the pressure is pretty large in NVT ensemble. Or the reason is due to the lack of enough numbers of atoms?

Here is the param.json:

> {
    "type_map": [
        "Si"
    ],
    "mass_map": [
        28.0855
    ],
    "init_data_prefix": "/home/web_server/MMCP/upload/1687272082.8427019",
    "init_data_sys": [
        "init_bulk/POSCAR.init_bulk.01x01x01/02.md/sys-0008/deepmd"
    ],
    "sys_configs_prefix": "/home/web_server/MMCP/upload/1687272082.8427019",
    "sys_configs": [
        [
            "init_bulk/POSCAR.init_bulk.01x01x01/01.scale_pert/sys-0008/scale-1.000/00000*/POSCAR"
        ],
        [
            "init_bulk/POSCAR.init_bulk.01x01x01/01.scale_pert/sys-0008/scale-1.000/00001*/POSCAR"
        ],
        [
            "init_bulk/POSCAR.init_bulk.01x01x01/01.scale_pert/sys-0008/scale-1.000/00002*/POSCAR"
        ]
    ],
    "numb_models": 4,
    "default_training_param": {
        "model": {
            "type_map": [
                "Si"
            ],
            "descriptor": {
                "type": "se_e2_a",
                "sel": "auto",
                "rcut_smth": 0.5,
                "rcut": 6.0,
                "neuron": [
                    120,
                    120,
                    120
                ],
                "resnet_dt": false,
                "axis_neuron": 16,
                "seed": 1
            },
            "fitting_net": {
                "neuron": [
                    25,
                    50,
                    100
                ],
                "resnet_dt": false,
                "seed": 1
            }
        },
        "learning_rate": {
            "type": "exp",
            "start_lr": 0.001,
            "decay_steps": 4000
        },
        "loss": {
            "start_pref_e": 0.02,
            "limit_pref_e": 2,
            "start_pref_f": 1000,
            "limit_pref_f": 1,
            "start_pref_v": 0.0,
            "limit_pref_v": 0.0
        },
        "training": {
            "stop_batch": 2000,
            "disp_file": "lcurve.out",
            "disp_freq": 1000,
            "numb_test": 4,
            "save_freq": 1000,
            "save_ckpt": "model.ckpt",
            "disp_training": true,
            "time_training": true,
            "profiling": false,
            "profiling_file": "timeline.json"
        }
    },
    "model_devi_dt": 0.002,
    "model_devi_skip": 0,
    "model_devi_f_trust_lo": 0.1,
    "model_devi_f_trust_hi": 0.3,
    "model_devi_e_trust_lo": 10000000000.0,
    "model_devi_e_trust_hi": 10000000000.0,
    "model_devi_clean_traj": true,
    "model_devi_jobs": [
        { "sys_idx": [ 0 ], "temps": [200, 300,400,500,600 ],"press": [1.0],"trj_freq": 20, "nsteps": 1000, "ensemble": "nvt", "_idx": "00"},
        {"sys_idx": [ 0 ], "temps": [200,300,400,500,600 ],"press": [1.0,50.0,100.0],"trj_freq": 20,"nsteps": 2000,"ensemble": "npt","_idx": "01"}
    ],
    "fp_style": "vasp",
    "shuffle_poscar": false,
    "fp_task_max": 20,
    "fp_task_min": 5,
    "fp_pp_path": "./",
    "fp_pp_files": [
        "POTCAR_Si"
    ],
    "fp_incar": "./INCAR.scf"
}

**Comment by njzjz** [comment]
Please check if https://github.com/deepmodeling/deepmd-kit/discussions/1622 is helpful.
